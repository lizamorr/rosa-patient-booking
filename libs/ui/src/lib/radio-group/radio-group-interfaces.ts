export interface IRadio {
  label: string;
  input: IRadioInput[];
}

interface IRadioInput {
  type: string;
  label: string;
  value: string;
  name: string;
  id: string;
}
