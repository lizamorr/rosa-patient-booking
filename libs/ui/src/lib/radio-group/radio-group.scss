.radio-group {
  flex: 1;
  display: flex;
  flex-direction: column;
  padding: 16px 32px;
  &__label {
    font-size: 1rem;
    color: #594d49;
    padding-bottom: 4px;
    margin: 0;
  }
  &__input-container {
    display: flex;
    flex-direction: row;
    margin: 0;
    padding: 0;
    border: 0;
    justify-content: flex-start;
    box-sizing: border-box;
  }
  &__input {
    width: 100%;
    margin-right: 0 !important;
    display: flex;
    align-items: center;
    border: 1px solid #b8afac;
  }
  &__input:not(:last-child) {
    border-right: 0;
  }
  input[type='radio'] {
    appearance: none;
    margin: 0;
  }
  input[type='radio'] + *::before {
    left: 0;
    background-color: #fff;
    margin: 0 0 0 12px;
    content: '';
    position: absolute;
    font: inherit;
    width: 12px;
    height: 12px;
    border: 2px solid #7f5bc2;
    border-radius: 50%;
    background: linear-gradient(
      180deg,
      rgba(0, 0, 0, 0.15) 0%,
      rgba(196, 196, 196, 0) 50%
    );
  }
  input[type='radio']:checked + * {
    color: #7f5bc2;
  }
  input[type='radio']:checked + *::before {
    background: radial-gradient(
      #7f5bc2 0%,
      #7f5bc2 50%,
      transparent 50%,
      transparent
    );
  }
  label {
    position: relative;
    cursor: pointer;
    line-height: 1rem;
    display: block;
    width: 100%;
    box-sizing: border-box;
    margin: 0;
    padding: 12px;
    padding-left: 36px !important;
  }
}
