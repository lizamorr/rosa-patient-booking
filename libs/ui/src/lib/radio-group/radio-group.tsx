import React from 'react';
import './radio-group.scss';

interface RadioGroupProps {
  radioInput: RadioGroupInput[];
  radioLabel: string;
  radioAnswer: string;
  onRadioChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
}
export interface RadioGroupInput {
  type: string;
  name: string;
  value: string;
  id?: string;
  label: string;
  placeholder?: string;
  required?: boolean;
}

export function RadioGroup(props: RadioGroupProps) {
  const { radioInput, radioLabel, onRadioChange, radioAnswer } = props;
  return (
    <div className="radio-group">
      {radioLabel && <span className="radio-group__label">{radioLabel}</span>}
      <div className="radio-group__input-container">
        {radioInput.map((input, index) => (
          <div key={index} className="radio-group__input">
            <input
              key={index}
              type={input.type}
              name={input.name}
              id={input.id}
              value={input.value}
              checked={input.value === radioAnswer}
              required={input.required}
              onChange={onRadioChange}
            />
            {input.id && input.label && (
              <label htmlFor={input.id}>{input.label}</label>
            )}
          </div>
        ))}
      </div>
    </div>
  );
}

export default RadioGroup;
