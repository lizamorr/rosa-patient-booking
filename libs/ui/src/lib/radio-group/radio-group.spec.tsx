import { render } from '@testing-library/react';

import { RadioGroup } from './radio-group';

describe('RadioGroup', () => {
  const radioProps = {
    radioLabel: 'Is this your first appointment with this practitioner?',
    radioAnswer: 'Yes',
    onRadioChange: jest.fn(),
    radioInput: [
      {
        type: 'radio',
        label: 'Yes',
        value: 'yes',
        name: 'patientStatus',
        id: 'input-radio__yes',
      },
      {
        type: 'radio',
        label: 'No',
        value: 'no',
        name: 'patientStatus',
        id: 'input-radio__no',
      },
    ],
  };
  it('should match snapshot', () => {
    const component = render(<RadioGroup {...radioProps} />);
    expect(component).toMatchSnapshot();
  });
});
