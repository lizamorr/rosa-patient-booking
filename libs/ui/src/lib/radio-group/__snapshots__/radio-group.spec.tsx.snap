// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioGroup should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="radio-group"
      >
        <span
          class="radio-group__label"
        >
          Is this your first appointment with this practitioner?
        </span>
        <div
          class="radio-group__input-container"
        >
          <div
            class="radio-group__input"
          >
            <input
              id="input-radio__yes"
              name="patientStatus"
              type="radio"
              value="yes"
            />
            <label
              for="input-radio__yes"
            >
              Yes
            </label>
          </div>
          <div
            class="radio-group__input"
          >
            <input
              id="input-radio__no"
              name="patientStatus"
              type="radio"
              value="no"
            />
            <label
              for="input-radio__no"
            >
              No
            </label>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="radio-group"
    >
      <span
        class="radio-group__label"
      >
        Is this your first appointment with this practitioner?
      </span>
      <div
        class="radio-group__input-container"
      >
        <div
          class="radio-group__input"
        >
          <input
            id="input-radio__yes"
            name="patientStatus"
            type="radio"
            value="yes"
          />
          <label
            for="input-radio__yes"
          >
            Yes
          </label>
        </div>
        <div
          class="radio-group__input"
        >
          <input
            id="input-radio__no"
            name="patientStatus"
            type="radio"
            value="no"
          />
          <label
            for="input-radio__no"
          >
            No
          </label>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
